client/src/components/Attachment/UploadAttachment.js

```diff
@@ -0,0 +1,167 @@
+    selectedAttachment.attachmentRelatedObjects[0].relatedObjectType = type
+    selectedAttachment.attachmentRelatedObjects[0].relatedObjectUuid = null
+    selectedAttachment.classification =
+      Settings.fields.attachment.classification.choices.UNDEFINED.value
+
+    save(selectedAttachment).then(response => {
+      selectedAttachment.uuid = response.createAttachment
+      selectedAttachment.contentLength = file.size
+      handleUploadFile(selectedAttachment)
+    })
+  }
+
+  // Convert file to base64 string
+  const convertBase64 = file => {
+    return new Promise((resolve, reject) => {
+      const fileReader = new FileReader()
+      fileReader.readAsDataURL(file)
+
+      fileReader.onload = () => {
+        resolve(fileReader.result)
+      }
+
+      fileReader.onerror = error => {
+        reject(error)
+      }
+    })
+  }
+
+  return (
+    <div>
+      {error && <Messages error={error} />}
+      {/** **** Select and drop file in here **** **/}
+      <section className="FileUploadContainer">
+        <div>
+          <p className="DragDropText">
+            <Icon className="uploadIcon" size={30} icon={IconNames.EXPORT} />
+            <span style={{ marginTop: "5px" }}>
+              <b>Choose a file</b> or Drag it here
+            </span>
+          </p>
+        </div>
+        <input
+          className="FormField"
+          id="fileUpload"
+          type="file"
+          accept="image/*, .pdf"
+          onChange={handleFileEvent}
+        />
+      </section>
+
+      {/** **** Show uploaded files in here **** **/}
+      <div style={{ display: "flex", flexWrap: "wrap" }}>
+        {uploadedList.map((file, index) => (
+          <AttachmentCard
+            key={index}
+            file={file}
+            index={index}
+            remove={remove}
+            setError={setError}
+            setRemove={setRemove}
+            uploadedList={uploadedList}
+            setUploadedList={setUploadedList}
+          />
+        ))}
+        {/** For edit report edit page show uploaded attachments **/}
+        {edit && <UploadedAttachments uuid={relatedObjectUuid} />}
+      </div>
+    </div>
+  )
+
+  function onSubmit(relatedObjectUuid) {
+    const unsavedAttachment = uploadedList.filter(
+      attach =>
+        !attach.attachmentRelatedObjects[0].relatedObjectUuid &&
+        attach.uuid !== null
+    )
+    if (unsavedAttachment.length > 0) {
+      for (let i = 0; i < unsavedAttachment.length; i++) {
+        save(unsavedAttachment[i], relatedObjectUuid, true).catch(() => {
+          setLoadingError(true)
+        })
+      }
+    }
+  }
+
+  function save(value, relatedObjectUuid, edit) {
+    const attachment = Attachment.filterClientSideFields(value)
+    const base64Index =
+      value.content?.indexOf(base64Marker) + base64Marker.length
+
+    attachment.content = value.content?.substring(base64Index)
+    attachment.attachmentRelatedObjects[0].relatedObjectUuid = relatedObjectUuid
```

Comment:
This should be unnecessary if you can only upload attachments for existing objects. In any case this will fail once attachments can be linked to multiple objects.